.github/workflows/run-tests.yml

```diff
@@ -27,11 +27,12 @@ jobs:
           python-version: ${{ matrix.python-version }}
       - name: Install dependencies
         run: |
-          python -m pip install --upgrade pip
-          pip install -r requirements.txt
-      - name: Run nosetests
-        run: |
-          nosetests
+          python -m pip install --upgrade pip poetry
+          pip install ".[dev]"
```

Comment:
We're using pip to install the dependencies instead of poetry for a few reasons:
- We want to support pip installations. This will probably be the primary way
- A deployment doesn't actually need poetry, which would add another dependency. Use poetry primarily for developing.
- Poetry has its own system of managing environments, and where packages are installed by default, which can get complex. https://realpython.com/dependency-management-python-poetry/ eg. When trying to run the CI earlier with poetry, `poetry install` would install the packages (including the test executable) into `~/.cache` which weren't automatically usable by the system installed Python (causing all those test failures). We don't need to deal with this.